app/process/watcher/evm/watcher.go

```diff
@@ -486,8 +516,17 @@ func (ew *Watcher) handleLockLog(eventLog *router.RouterLock, q qi.Queue) {
 			return
 		}
 	}
-	if properAmount.Cmp(big.NewInt(0)) == 0 {
-		ew.logger.Errorf("[%s] - Insufficient amount provided: Event Amount [%s] and Proper Amount [%s].", eventLog.Raw.TxHash, eventLog.Amount, properAmount)
+
+	nativeAsset := ew.assetsService.FungibleNativeAsset(sourceChainId, token)
+	tokenPriceInfo, exist := ew.pricingService.GetTokenPriceInfo(sourceChainId, nativeAsset.Asset)
+	if !exist {
+		ew.logger.Errorf("[%s] - Couldn't get price info in USD for asset [%s].", eventLog.Raw.TxHash, nativeAsset.Asset)
+		return
+	}
+
+	amountInUsd := decimalHelper.GetAmountInUsd(tokenPriceInfo.UsdPrice, properAmount, ew.assetsService, nativeAsset)
+	if amountInUsd.Cmp(tokenPriceInfo.MinAmountInUsdWithFee) < 0 {
+		ew.logger.Errorf("[%s] - Transfer Amount [%s] less than Minimum Amount [%s].", eventLog.Raw.TxHash, amountInUsd, nativeAsset.MinFeeAmountInUsd)
```

Comment:
Wouldn't it be better to only compare amounts here rather than doing calculations every time?

Reply:
Yep fixing it now.